container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java

```diff
@@ -18,6 +23,15 @@
     
     private final QueryProfileTypeRegistry typeRegistry;
 
+    @Inject
+    public CompiledQueryProfileRegistry(QueryProfilesConfig config) {
+        QueryProfileRegistry registry = QueryProfileConfigurer.createFromConfig(config);
+        typeRegistry = registry.getTypeRegistry();
+        for (QueryProfile inputProfile : registry.allComponents()) {
```

Comment:
Why not just do 
QueryProfileCompiler.compile(registry)
here?

Reply:
I didn't want to create a temporary object of the same type as the class itself. There is room for improvement, but it requires creating even more public methods and/or constructors.